app/spec/lib/bk/compat/circleci_spec.rb

```diff
@@ -21,3 +21,143 @@
+      end
+
+      it 'uses default installation method with custom directory' do
+        result = installer.install_commands
+
+        expect(result).to include('curl -fsSL https://get.docker.com -o get-docker.sh')
+        expect(result).to include('sh get-docker.sh')
+      end
+    end
+
+    context 'with version parameter' do
+      let(:installer) { described_class.new({ 'version' => 'v20.10.23' }) }
+
+      it 'installs specific version' do
+        result = installer.install_commands
+
+        expect(result).to include('VERSION=v20.10.23')
+        expect(result).to include(
+          'curl -fsSL https://download.docker.com/linux/static/stable/x86_64/' \
+          'docker-${VERSION}.tgz -o docker.tgz'
+        )
+        expect(result).to include('tar xzvf docker.tgz')
+        expect(result).to include('rm -rf docker docker.tgz')
+      end
+
+      it 'uses default installation directory' do
+        result = installer.install_commands
+
+        expect(result).to include('$SUDO cp docker/* /usr/local/bin/')
+      end
+    end
+
+    context 'with both version and install-dir parameters' do
+      let(:installer) do
+        described_class.new({
+                              'version' => 'v20.10.23',
+                              'install-dir' => '/opt/docker'
+                            })
+      end
+
+      it 'combines both custom directory and version installation' do
+        result = installer.install_commands
+
+        expect(result).to include('$SUDO mkdir -p /opt/docker')
+        expect(result).to include('VERSION=v20.10.23')
+        expect(result).to include('$SUDO cp docker/* /opt/docker/')
+      end
+    end
+
+    context 'with default parameter values' do
+      let(:installer) do
+        described_class.new({
+                              'install-dir' => '/usr/local/bin',
+                              'version' => 'latest'
+                            })
+      end
+
+      it 'uses simple installation when defaults are specified explicitly' do
+        result = installer.install_commands
+
+        expect(result).to eq([
+                               "echo '~~~ Installing Docker'",
+                               'curl -fsSL https://get.docker.com -o get-docker.sh',
+                               'sh get-docker.sh'
+                             ])
+      end
+    end
+
+    context 'with invalid parameters' do
+      let(:installer) { described_class.new({ 'unknown' => 'value' }) }
+
+      it 'ignores unknown parameters' do
+        result = installer.install_commands
+
+        expect(result).to eq([
+                               "echo '~~~ Installing Docker'",
+                               'curl -fsSL https://get.docker.com -o get-docker.sh',
+                               'sh get-docker.sh'
+                             ])
+      end
+    end
+  end
+end
+
+RSpec.describe BK::Compat::CircleCISteps::DockerOrb do
```

Comment:
there is no need to test this class specifically, the snapshot testing takes care of that